src/vSphereVirtualMachineCollector.class.inc.php

```diff
@@ -317,7 +317,7 @@ static protected function DoCollectVMInfo($aFarms, $oVirtualMachine, $aVLANs, $i
 		} else {
 			// ManagementIP cannot be an IPV6 address, if no IPV4 was found above, let's clear the field
 			// Note: some OpenVM clients report IP addresses with a trailing space, so let's trim the field
-			$aData['managementip'] = (strpos($sGuestIP, ':') !== false) ? '' : trim($sGuestIP);
+			$aData['managementip'] = (filter_var($sGuestIP, FILTER_VALIDATE_IP, FILTER_FLAG_IPV6)) ? '' : trim($sGuestIP);
```

Comment:
This will fail when the IP has a leading space as noted in the comments: https://3v4l.org/VH3Cu

Probably needs some fixing in other places as well. I suggest to trim the var when assigning `$sGuestIP`.

Reply:
Thanks for pointing out to that comment. I will rewrite the code

Reply:
done